### PR TITLE
fix(self-hosted): remove mfa factor is broken

### LINKED ISSUE #17680: Remove MFA Factors is broken in local Studio
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Clicking on "Remove MFA Factors" in local Studio throws an error. Action does not complete.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to `http://localhost:54323/project/default/auth/users`
2. Click on the button with three dots on any user
3. Click "Remove MFA Factors"

## Expected behavior

Remove MFA Factors for the user

## Screenshots

Screenshot 2023-09-24 at 9 17 07 AM

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of supabase-js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

## Additional context

```Failed to delete user MFA factors: <!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><title>Supabase</title><meta name="viewport" content="initial-scale=1.0, width=device-width"/><link rel="apple-touch-icon-precomposed" sizes="57x57" href="/favicon/apple-touch-icon-57x57.png"/><link rel="apple-touch-icon-precomposed" sizes="114x114" href="/favicon/apple-touch-icon-114x114.png"/><link rel="apple-touch-icon-precomposed" sizes="72x72" href="/favicon/apple-touch-icon-72x72.png"/><link rel="apple-touch-icon-precomposed" sizes="144x144" href="/favicon/apple-touch-icon-144x144.png"/><link rel="apple-touch-icon-precomposed" sizes="60x60" href="/favicon/apple-touch-icon-60x60.png"/><link rel="apple-touch-icon-precomposed" sizes="120x120" href="/favicon/apple-touch-icon-120x120.png"/><link rel="apple-touch-icon-precomposed" sizes="76x76" href="/favicon/apple-touch-icon-76x76.png"/><link rel="apple-touch-icon-precomposed" sizes="152x152" href="/favicon/apple-touch-icon-152x152.png"/><link rel="icon" type="image/png" href="/favicon/favicon-196x196.png" sizes="196x196"/><link rel="icon" type="image/png" href="/favicon/favicon-96x96.png" sizes="96x96"/><link rel="icon" type="image/png" href="/favicon/favicon-32x32.png" sizes="32x32"/><link rel="icon" type="image/png" href="/favicon/favicon-16x16.png" sizes="16x16"/><link rel="icon" type="image/png" href="/favicon/favicon-128.png" sizes="128x128"/><meta name="application-name" content=" "/><meta name="msapplication-TileColor" content="#1E1E1E"/><meta name="msapplication-TileImage" content="mstile-144x144.png"/><meta name="msapplication-square70x70logo" content="mstile-70x70.png"/><meta name="msapplication-square150x150logo" content="mstile-150x150.png"/><meta name="msapplication-wide310x150logo" content="mstile-310x150.png"/><meta name="msapplication-square310x310logo" content="mstile-310x310.png"/><meta name="theme-color" content="#1E1E1E"/><link rel="shortcut icon" href="/favicon/favicon.ico"/><link rel="icon" type="image/png" href="/favicon/favicon.ico"/><link rel="manifest" href="/favicon/site.webmanifest"/><link rel="alternate" type="application/rss+xml" href="/feed.xml"/><link rel="apple-touch-icon" href="/favicon/favicon.ico"/><meta name="msapplication-config" content="/favicon/browserconfig.xml"/><meta name="next-head-count" content="30"/><link rel="stylesheet" type="text/css" data-name="vs/editor/editor.main" href="https://cdn.jsdelivr.net/npm/monaco-editor@0.37.0/min/vs/editor/editor.main.css"/><link rel="stylesheet" type="text/css" href="/css/fonts.css"/><link rel="preload" href="/_next/static/css/24862156fbf6a92b.css" as="style"/><link rel="stylesheet" href="/_next/static/css/24862156fbf6a92b.css" data-n-g=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js"></script><script defer="" src="/_next/static/chunks/9411-866ffac3c178cdb3.js"></script><script defer="" src="/_next/static/chunks/1213-d5e543bd24b62cff.js"></script><script defer="" src="/_next/static/chunks/2621-2533718c36324bc8.js"></script><script defer="" src="/_next/static/chunks/7606-847615b8cb82a463.js"></script><script defer="" src="/_next/static/chunks/5512.90fe7a453c6d7791.js"></script><script defer="" src="/_next/static/chunks/5652.641b8a11fadaeb97.js"></script><script src="/_next/static/chunks/webpack-5037b67e32c966db.js" defer=""></script><script src="/_next/static/chunks/framework-88c3fab5a4d30eea.js" defer=""></script><script src="/_next/static/chunks/main-3ca6fb5daa80fdb7.js" defer=""></script><script src="/_next/static/chunks/pages/_app-0ece61f4ad55c89e.js" defer=""></script><script src="/_next/static/chunks/pages/404-2f82b58972bef958.js" defer=""></script><script src="/_next/static/Qyo0qAft8R45jm5Bzir6I/_buildManifest.js" defer=""></script><script src="/_next/static/Qyo0qAft8R45jm5Bzir6I/_ssgManifest.js" defer=""></script></head><body class="dark"><div id="__next" data-reactroot=""><div class="min-h-full flex flex-col"><div class="relative mx-auto flex h-screen w-full flex-col items-center justify-center"><div class="absolute top-0 mx-auto w-full max-w-7xl px-8 pt-6 sm:px-6 lg:px-8"><nav class="relative flex items-center justify-between sm:h-10"><div class="flex flex-shrink-0 flex-grow items-center lg:flex-grow-0"><div class="flex w-full items-center justify-between md:w-auto"><a href="/projects"><span style="box-sizing:border-box;display:inline-block;overflow:hidden;width:initial;height:initial;background:none;opacity:1;border:0;margin:0;padding:0;position:relative;max-width:100%"><span style="box-sizing:border-box;display:block;width:initial;height:initial;background:none;opacity:1;border:0;margin:0;padding:0;max-width:100%"></span>supabase<noscript>supabase</noscript></span></a></div></div></nav></div><div class="absolute select-none opacity-[5%] filter transition duration-200 blur-none"><h1 style="font-size:28rem">404</h1></div><div class="flex flex-col items-center justify-center space-y-6 transition opacity-0"><div class="flex w-[380px] flex-col items-center justify-center space-y-3 text-center"><h3 class="text-xl">Looking for something? 🔍</h3><p class="text-scale-1100">We couldn&#x27;t find the page that you&#x27;re looking for!</p></div><div class="flex items-center space-x-4"><a href="/projects"><button type="button" class="relative justify-center cursor-pointer inline-flex items-center space-x-2 text-center font-regular ease-out duration-200 rounded-md outline-none transition-all outline-0 focus-visible:outline-4 focus-visible:outline-offset-1 border bg-brand-600 dark:bg-brand/70 hover:bg-brand-600/80 dark:hover:bg-brand text-white border-brand dark:border-brand focus-visible:outline-brand-600 shadow-sm text-sm leading-4 px-3 py-2"> <span class="truncate">Head back</span> </button></a></div></div></div></div><div></div></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{}},"page":"/404","query":{},"buildId":"Qyo0qAft8R45jm5Bzir6I","nextExport":true,"autoExport":true,"isFallback":false,"dynamicIds":[15652],"scriptLoader":[]}</script></body></html>```

### PR DESCRIPTION
**Pull request by tranhoangvuit**
## What kind of change does this PR introduce?

fixes: [17680](https://github.com/supabase/supabase/issues/17680)

## What is the current behavior?

After create a user, we remove MFA factors will return 404 error.

## What is the new behavior?

.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Review by kiwicopple — APPROVED** [review]

File: apps/studio/data/auth/user-delete-mfa-factors-mutation.ts [hunk 1]
```diff
@@ -11,7 +11,11 @@ export type UserDeleteMFAFactorsVariables = {
 }
 
 export async function deleteMFAFactors({ projectRef, userId }: UserDeleteMFAFactorsVariables) {
-  const response = await delete_(`${API_URL}/auth/${projectRef}/users/${userId}/factors`)
+  const response = IS_PLATFORM
+    ? await delete_(`${API_URL}/auth/${projectRef}/users/${userId}/factors`)
```
Comment (tranhoangvuit) [inline apps/studio/data/auth/user-delete-mfa-factors-mutation.ts:15]: On Supabase Cloud we call to this endpoint: 
`https://api.supabase.io/platform/auth/jnyalqqtavnsmjgplrzt/users/00eb4686-372e-4696-9db0-8a6e0be8294a/factors`
But on local:
`http://localhost:8082/api/auth/default/users/00eb4686-372e-4696-9db0-8a6e0be8294a/factors`
I'm unsure whether this change is correct, and I also have uncertainties regarding the logic on the factors endpoint.

**Comment by alaister** [comment]
Hey @tranhoangvuit,

Thanks for the PR!

I think a better way of solving this is to add the missing local API route. There's many different examples of proxy endpoints in the `studio/pages/api` folder to follow.

Thanks again 🙌

**Comment by tranhoangvuit** [comment]
Thank @kiwicopple, @alaister . Can I confirm the logic behind removing the MFA factor? Is it as follows:
- We will add a local API with the same URL as the `platform` (which is currently missing).
- We will remove the record from the `mfa_factors` table that belongs to the `user_id`.
Is this correct?

**Comment by alaister** [comment]
Hey @tranhoangvuit,

Sure! Step 1 is correct, the missing route will need to be created locally in the `studio/pages/api` directory. For step 2, you shouldn't have to manually remove the factor from the mfa_factors table. You should just be able to send a call to the local gotrue instance to do this for you :)

**Review by alaister — APPROVED** [review]
LGTM! Thanks, @tranhoangvuit!

**Pull request merged**